master/pkg/logger/log_buffer.go

```diff
@@ -188,3 +188,23 @@ func LogrusLevelToProto(l logrus.Level) logv1.LogLevel {
 		panic("invalid logrus log level")
 	}
 }
+
+// ProtoToLogrusLevel translates our protobuf log level to a logrus level.
+func ProtoToLogrusLevel(l logv1.LogLevel) logrus.Level {
+	switch l {
+	case logv1.LogLevel_LOG_LEVEL_TRACE:
+		return logrus.TraceLevel
+	case logv1.LogLevel_LOG_LEVEL_DEBUG:
+		return logrus.DebugLevel
+	case logv1.LogLevel_LOG_LEVEL_INFO:
+		return logrus.InfoLevel
+	case logv1.LogLevel_LOG_LEVEL_WARNING:
+		return logrus.WarnLevel
+	case logv1.LogLevel_LOG_LEVEL_ERROR:
+		return logrus.ErrorLevel
+	case logv1.LogLevel_LOG_LEVEL_CRITICAL:
+		return logrus.FatalLevel
+	default:
+		panic("invalid logrus log level")
```

Comment:
question (non-blocking): would it be worth returning an error here instead of panic-ing? Or is this state non-recoverable, and we need terminate?

Reply:
I think an error could work too, but I think for consistency I would change it in LogrusLevelToProto as well. What do you think?

Reply:
i'm a fan of panic in this case, since it is definitely a bug and should never happen. if you want, there is a linter you can turn on to make sure your enum type switches are exhaustive. iirc its available as a sublinter of golangci-lint https://pkg.go.dev/github.com/nishanths/exhaustive